### PR TITLE
Introduce dynamic plugin SDK build root image that uses node 16

### PR DESCRIPTION
**Pull request by florkbr**
This will be PR 1 of 2 to switch the dynamic-plugin-sdk PR jobs/config over to using a more up to date base image (node 16). 

This PR introduces the new image to be built/published to the CI namespace. The next PR will update the SDK to switch over to it.

### CODE REVIEW

File: clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  successfulBuildsHistoryLimit: 1
+  failedBuildsHistoryLimit: 2
+  output:
+    to:
+      kind: ImageStreamTag
+      name: openshift-dynamic-plugin-sdk-buildroot:nodejs16
+      namespace: ci
+  source:
+    dockerfile: |
+      FROM quay.io/centos/centos:stream8
```
Comment (florkbr) [inline clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:22]: @bear-redhat @vojtechszocs here is the result from the build of this dockerfile locally: 
[dynamic-plugin-sdk-buildroot-docker-build.log](https://github.com/openshift/release/files/9569600/dynamic-plugin-sdk-buildroot-docker-build.log)

File: clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          yum install -y $INSTALL_PKGS && \
+          rpm -V $INSTALL_PKGS && \
+          yum clean all && \
+          touch /os-build-image && \
+          git config --system user.name origin-release-container && \
+          git config --system user.email origin-release@redhat.com && \
+          mkdir $WORKDIR
+
+      ENV VAULT_DISABLE_MLOCK true
+      RUN yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo && \
```
Comment (florkbr) [inline clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]: @bear-redhat can you chime in if there is documentation around which of these packages is required? @vojtechszocs and I noticed `mercurial` is installed above and we likely don't need it for example.

Comment (karelhala) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]: We could reuse this docker image https://github.com/RedHatInsights/insights-frontend-builder-common/blob/master/src/releaser.Dockerfile We could add some additional dependencies if needed.

Comment (florkbr) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]: @vojtechszocs what is your opinion here? I'm fine with either approach TBH, though I agree we should converge on a single "nodejs-16" image for our projects.

Comment (vojtechszocs) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]: The Dockerfile @karelhala referred to includes some things we don't need, such as `golang` package and `github.com` ssh host config.

I'd suggest to go with a minimal Dockerfile that provides
- explicit control over which Node.js version is installed (don't rely on what is provided by default)
- meets the minimum filesystem & package requirements of OpenShift `ci-operator` [1]
  - installed `git` package
  - writable `/go` directory
- npm package manager updated to latest stable version
- Yarn installed as a global npm package

[1] https://docs.ci.openshift.org/docs/architecture/ci-operator/#build-root-image

Comment (vojtechszocs) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]:
```Dockerfile
FROM quay.io/fedora/fedora:36-x86_64

RUN dnf update -y && \
    dnf install git jq -y && \
    git config --system user.name origin-release-container && \
    git config --system user.email origin-release@redhat.com && \
    dnf module enable nodejs:16 -y && \
    dnf module install nodejs:16 -y && \
    WORKDIR='/go' && \
    mkdir $WORKDIR && \
    npm config -g set cache $WORKDIR/.npm && \
    npm install -g --no-update-notifier npm && \
    npm install -g --no-update-notifier yarn && \
    dnf clean all
```

Comment (vojtechszocs) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml:38]: I've tested the above Dockerfile - it uses Node.js `v16.15.0` and npm `8.19.2`

```
$ docker images 
REPOSITORY              TAG         IMAGE ID       CREATED          SIZE
test                    latest      1da3243d7710   14 minutes ago   424MB
quay.io/fedora/fedora   36-x86_64   2ecb6df95994   8 weeks ago      163MB
```

The image size is relatively small (424MB) compared to the image based on CentOS 8 :smiley: 

The build log is here: [Test.dockerfile.log](https://github.com/openshift/release/files/9577951/Test.dockerfile.log)

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

File: clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+metadata:
+  name: openshift-dynamic-plugin-sdk-buildroot
+  namespace: ci
+spec:
+  successfulBuildsHistoryLimit: 1
+  failedBuildsHistoryLimit: 2
+  output:
+    to:
+      kind: ImageStreamTag
+      name: openshift-dynamic-plugin-sdk-buildroot:nodejs16
```
[the remark this answers is not in the document]
Comment (florkbr) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml]: Done!

File: clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml (hunk 2 above)
Comment (florkbr) [reply clusters/app.ci/supplemental-ci-images/dynamic-plugin-sdk-buildroot.yaml]: Done!

**Comment by florkbr** [comment]
@vojtechszocs implemented your new suggestions. Logs: 
[dynamic-plugin-sdk-docker-build.log](https://github.com/openshift/release/files/9578269/dynamic-plugin-sdk-docker-build.log)

**Pull request merged**